cfpb/cfgov-refresh
# Updates pagination to support on-demand generation + stepper

Updates pagination to support on-demand generation and adds a stepper variant as mentioned in [GHE]/flapjack/Modules-V1/issues/122

## Changes

- Updated pagination to support "stepper" modifier.
- Updated pagination to work in on-demand workflow.
- Updated pagination to support multiple pagination molecules on a single page.
- Moved pagination tests from /organisms/ to /molecules/ where they belong.
- Converted `m-pagination_submit-btn` id to class. 

## Testing

- `gulp styles:ondemand`
- visit /test-fixture/?atomic=pagination to see regular pagination component.
- visit /test-fixture/?atomic=pagination.is_stepper to see stepper variant.
- visit /blog/ and check the pagination with and without filter work correctly.
- `gulp test:acceptance --sauce=false` should pass.

## Review

- @kurtw 
- @Scotchester 
- @sebworks 
- @jimmynotjim 
- @KimberlyMunoz 

## Screenshots

<img width="270" alt="screen shot 2016-06-17 at 2 45 10 pm" src="https://cloud.githubusercontent.com/assets/704760/16200300/c070cec0-36da-11e6-8be2-96f32d0db54f.png">

<img width="269" alt="screen shot 2016-06-17 at 2 45 21 pm" src="https://cloud.githubusercontent.com/assets/704760/16200301/c0724d9a-36da-11e6-85e0-8c32fa0c49b2.png">

<img width="855" alt="screen shot 2016-06-17 at 2 45 34 pm" src="https://cloud.githubusercontent.com/assets/704760/16200302/c07428c2-36da-11e6-99e1-a5072eb853a5.png">

<img width="855" alt="screen shot 2016-06-17 at 2 45 44 pm" src="https://cloud.githubusercontent.com/assets/704760/16200303/c076d00e-36da-11e6-9288-5f9ec14ee0b3.png">
